src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    /// <summary>
+    /// Wrapper around a gss_name_t_desc*
+    /// </summary>
+    internal sealed class SafeGssNameHandle : SafeHandle
+    {
+        public static SafeGssNameHandle CreateUser(string name)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name), "Invalid user name passed to SafeGssNameHandle create");
+            SafeGssNameHandle retHandle;
+            Interop.NetSecurityNative.Status minorStatus;
+            Interop.NetSecurityNative.Status status = Interop.NetSecurityNative.ImportUserName(
+                out minorStatus, name, Encoding.UTF8.GetByteCount(name), out retHandle);
```

Comment:
Ah, good, I was unsure, and that's because the answer is "it's complicated".  Probably makes sense to leave this as-is, then.